tools/bumper/bumper.go

```diff
@@ -0,0 +1,194 @@
+			component.Metadata = updatedReleaseTag
+			err = updateComponentsYaml(inputArgs.componentsConfigPath, componentsConfig)
+			if err != nil {
+				exitWithError(errors.Wrap(err, "Failed to update components yaml"))
+			}
+
+			// run bump script
+			cmd := exec.Command("make", fmt.Sprintf("bump-%s", componentName))
+			if out, err := cmd.CombinedOutput(); err != nil {
+				exitWithError(errors.Wrapf(err, "Failed to run bump script. StdOut = %s", string(out)))
+			}
+
+			//create a new branch name
+			BranchName := strings.Replace(strings.ToLower(PRTitle), " ", "_", -1)
+			logger.Printf("Opening new Branch %s", BranchName)
+
+			//push branch to PR
+			exitWithError(fmt.Errorf("push branch to PR not implemented yet"))
+		}
+	}
+}
+
+func getRepositoryOnBranch(repoDir, repositoryURL string, branchName string) (*git.Repository, error) {
+	logger.Printf("Cloning to temp directory: %s", repoDir)
+
+	return git.PlainClone(repoDir, false, &git.CloneOptions{
+		URL:           repositoryURL,
+		ReferenceName: plumbing.NewBranchReferenceName(branchName),
+	})
+}
+
+func isBumpNeeded(currentReleaseVersion, latestReleaseVersion, updatePolicy string) (bool, error) {
+	logger.Printf("currentReleaseVersion: %s, latestReleaseVersion: %s, updatePolicy: %s\n", currentReleaseVersion, latestReleaseVersion, updatePolicy)
+
+	if updatePolicy == updatePolicyStatic {
+		logger.Printf("updatePolicy is static. avoiding auto bump")
+		return false, nil
+	}
+
+	//if one of the tags is in vtag format (e.g 0.39.0-32-g1fcbe815), and not equal, then always bump
+	if isVtagFormat(currentReleaseVersion) || isVtagFormat(latestReleaseVersion) {
+		return currentReleaseVersion == latestReleaseVersion, nil
+	} else {
+		currentVersion, err := refactorVersion(currentReleaseVersion)
+		if err != nil {
+			return false, errors.Wrapf(err, "Failed to digest current Version %s to semver", currentVersion)
+		}
+		LatestVersion, err := refactorVersion(latestReleaseVersion)
+		if err != nil {
+			return false, errors.Wrapf(err, "Failed to digest latest Version %s to semver", LatestVersion)
+		}
+
+		return currentVersion.LessThan(*LatestVersion), nil
+	}
+}
+
+func initLog() *log.Logger {
+	var buf bytes.Buffer
+	logger := log.New(&buf, "INFO: ", log.LstdFlags)
+	logger.SetOutput(os.Stdout)
+	return logger
+}
+
+func initFlags(paramArgs *inputParams) {
+	flag.StringVar(&paramArgs.componentsConfigPath, "config-path", "", "Full path to components yaml")
+	flag.StringVar(&paramArgs.gitToken, "token", "", "git Token")
+	flag.Parse()
+	if flag.NFlag() != 2 {
+		exitWithError(fmt.Errorf("Wrong Number of input parameters %d, should be 2. Use --help for usage", flag.NFlag()))
+	}
+}
+
+// since versioning of components can sometimes divert from semver standard, we do some refactoring
+func refactorVersion(version string) (*semver.Version, error) {
+	//remove starting "v" if exists
+	version = strings.TrimPrefix(version, "v")
+
+	//expand to 2 dotted format
```

Comment:
Please start all comments with `// ` (note the space after //)

Reply:
Done